Sources_v3/WebSocketClient/WebSocketClient.swift

```diff
@@ -107,17 +104,18 @@ class WebSocketClient {
         sessionConfiguration: URLSessionConfiguration,
         requestEncoder: RequestEncoder,
         eventDecoder: AnyEventDecoder,
-        eventMiddlewares: [EventMiddleware],
+        notificationCenter: EventNotificationCenter,
         reconnectionStrategy: WebSocketClientReconnectionStrategy = DefaultReconnectionStrategy(),
         environment: Environment = .init()
     ) {
         self.environment = environment
         self.connectEndpoint = connectEndpoint
         self.requestEncoder = requestEncoder
         self.sessionConfiguration = sessionConfiguration
-        middlewares = eventMiddlewares
         self.reconnectionStrategy = reconnectionStrategy
         self.eventDecoder = eventDecoder
+        self.notificationCenter = notificationCenter
+        self.notificationCenter.add(middleware: HealthCheckMiddleware(webSocketClient: self))
```

Comment:
It'd be great to add a test that will verify we add the middleware and the delegate is set.

Reply:
added in d274230163815e0e3c136eaff77d254452d7f2e3